File: DotNet/CesiumLanguageWriter/Advanced/CesiumNearFarScalarValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumNearFarScalarValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="NearFarScalar"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumNearFarScalarValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumNearFarScalarValuePropertyAdaptor<TFrom> : CesiumInterpolatableWriterAdaptor<TFrom, NearFarScalar>
        where TFrom : class, ICesiumNearFarScalarValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="NearFarScalar"/>.</param>
        /// <param name="writeSamplesCallback">The callback to write samples of type <see cref="NearFarScalar"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumNearFarScalarValuePropertyAdaptor([NotNull] TFrom parent,
                                                       [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, NearFarScalar> writeValueCallback,
                                                       [NotNull] CesiumWriterAdaptorWriteSamplesCallback<TFrom, NearFarScalar> writeSamplesCallback,
                                                       [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumNearFarScalarValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * Adapts a class that implements {@link ICesiumNearFarScalarValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@link NearFarScalar} values.
 * @param <TFrom> The class that implements {@link ICesiumNearFarScalarValuePropertyWriter} to adapt.
 */
@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
public class CesiumNearFarScalarValuePropertyAdaptor<TFrom extends ICesiumNearFarScalarValuePropertyWriter> extends CesiumInterpolatableWriterAdaptor<TFrom, NearFarScalar> {
    /**
    * Initializes a new instance.
    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link NearFarScalar}.
    * @param writeSamplesCallback The callback to write samples of type {@link NearFarScalar}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumNearFarScalarValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, NearFarScalar> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteSamplesCallback<TFrom, NearFarScalar> writeSamplesCallback, @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback);
    }
}